packages/key-readmodel-writer/src/keyConsumerServiceV2.ts

```diff
@@ -1,17 +1,69 @@
 import { KeyCollection } from "pagopa-interop-commons";
-import { AuthorizationEventEnvelopeV2 } from "pagopa-interop-models";
+import {
+  AuthorizationEventEnvelopeV2,
+  fromClientV2,
+  toReadModelClient,
+} from "pagopa-interop-models";
 import { match } from "ts-pattern";
 
 export async function handleMessageV2(
   message: AuthorizationEventEnvelopeV2,
-  _keys: KeyCollection
+  keys: KeyCollection
 ): Promise<void> {
-  match(message)
+  await match(message)
+    .with({ type: "ClientKeyAdded" }, async (message) => {
+      const client = message.data.client
+        ? toReadModelClient(fromClientV2(message.data.client))
+        : undefined;
+
+      const key = client?.keys.find((key) => key.kid === message.data.kid);
+
+      /*
+      to do: double-check version retrieval.
+      the implementation below retrieves the version number by reading the existing key (if any) in readmodel.
+      
+      Alternative implementation: we could use the version number of the client, but this means that the key will "skip" some versions.
+      For example, if a key is created at the 7th update of the client, this means that the key will be inserted in the keys collection with version 7
```

Comment:
In order to handle edge cases of the streaming platform (eventual consistency, message duplication from topics, etc) the version must be the one inside the event, as it is the only source of truth.

Plus, using the client version allows identifying the event that generated the specific document (in this case the key) in the read model

(the version "skip" is acceptable for our use case, as we do not need history, and we use it only to enforce the order of event processing)